packages/graphql-language-service-server/src/__tests__/MessageProcessor-test.ts

```diff
@@ -208,56 +208,154 @@ describe('MessageProcessor', () => {
     await expect(result[0].uri).toEqual(`file://${queryDir}/test3.graphql`);
   });
 
-  it('getQueryAndRange finds queries in tagged templates', async () => {
-    const text = `
-// @flow
-import {gql} from 'react-apollo';
-import type {B} from 'B';
-import A from './A';
-
-const QUERY = gql\`
-query Test {
-  test {
-    value
-    ...FragmentsComment
-  }
-}
-\${A.fragments.test}
-\`
-
-export function Example(arg: string) {}`;
-
-    const contents = getQueryAndRange(text, 'test.js');
-    expect(contents[0].query).toEqual(`
-query Test {
-  test {
-    value
-    ...FragmentsComment
-  }
-}
-`);
-  });
+  describe('getQueryRange', () => {
+    describe('JavaScript', () => {
+      function testExtension(ext: string) {
+        describe('Extension: ".' + ext + '"', () => {
+          it('finds queries in tagged templates', () => {
+            const text = `
+    import {gql} from 'react-apollo';
+    import type {B} from 'B';
+    import A from './A';
+    
+    const QUERY = gql\`
+    query Test {
+      test {
+        value
+        ...FragmentsComment
+      }
+    }
+    \${A.fragments.test}
+    \`
+    
+    export function Example(arg: string) {}`;
 
-  it('getQueryAndRange ignores non gql tagged templates', async () => {
-    const text = `
-// @flow
-import randomthing from 'package';
-import type {B} from 'B';
-import A from './A';
-
-const QUERY = randomthing\`
-query Test {
-  test {
-    value
-    ...FragmentsComment
-  }
-}
-\${A.fragments.test}
-\`
+            const contents = getQueryAndRange(text, 'test.' + ext);
+            expect(contents[0].query).toEqual(`
+    query Test {
+      test {
+        value
+        ...FragmentsComment
+      }
+    }
+    `);
+          });
+
+          it('ignores non gql tagged templates', () => {
+            const text = `
+    import randomthing from 'package';
+    import type {B} from 'B';
+    import A from './A';
+    
+    const QUERY = randomthing\`
+    query Test {
+      test {
+        value
+        ...FragmentsComment
+      }
+    }
+    \${A.fragments.test}
+    \`
+    
+    export function Example(arg: string) {}`;
+
+            const contents = getQueryAndRange(text, 'test.' + ext);
+            expect(contents.length).toEqual(0);
+          });
+        });
+      }
+
+      ['js', 'ts', 'jsx', 'tsx', 'es6', 'mjs', 'esm.js'].forEach(testExtension);
```

Comment:
nicely done!

for future reference. it.each() is great for this

also for future reference, I just wanted to test a single ts file to make sure we were set up to parse typescript as well, since this example still doesnt show a plain javascript example